src/lib/collectors/utils/requester.ts

```diff
@@ -41,11 +42,14 @@ export class Requester {
     private _request: request;
     /** Internal `redirectManager`. */
     private _redirects = new RedirectManager();
+    /** Maximum number of redirects */
+    private _limit;
 
-    constructor(customOptions = {}) {
+    constructor(customOptions = {}, redirectsLimit = 3) {
```

Comment:
Also we should make sure that if the user sets `followRedirect: true` in `customOptions` we set it to `false`.